packages/core/src/browser/view-container.ts

```diff
@@ -0,0 +1,935 @@
+        const body = document.createElement('div');
+        body.classList.add('body');
+        this.node.appendChild(header);
+        this.node.appendChild(body);
+        disposable.push(headerDisposable);
+        return {
+            header,
+            body,
+            disposable,
+        };
+    }
+
+    protected createHeader(): { header: HTMLElement, disposable: Disposable } {
+        const disposable = new DisposableCollection();
+        const header = document.createElement('div');
+        header.classList.add('theia-header', 'header');
+        disposable.push(addEventListener(header, 'click', () => {
+            this.collapsed = !this.collapsed;
+        }));
+
+        const toggleIcon = document.createElement('span');
+        toggleIcon.classList.add(EXPANSION_TOGGLE_CLASS);
+        if (this.collapsed) {
+            toggleIcon.classList.add(COLLAPSED_CLASS);
+        }
+        header.appendChild(toggleIcon);
+
+        const title = document.createElement('span');
+        title.classList.add('label', 'noselect');
+        title.innerText = this.wrapped.title.label;
+        header.appendChild(title);
+
+        if (ViewContainerPart.ContainedWidget.is(this.wrapped)) {
+            for (const { tooltip, execute, className } of this.wrapped.toolbarElements.filter(e => e.enabled !== false)) {
+                const toolbarItem = document.createElement('span');
+                toolbarItem.classList.add('element');
+                if (typeof className === 'string') {
+                    toolbarItem.classList.add(...className.split(' '));
+                } else if (Array.isArray(className)) {
+                    toolbarItem.classList.add(...className);
+                }
+                toolbarItem.title = tooltip;
+                disposable.push(addEventListener(toolbarItem, 'click', async event => {
+                    event.stopPropagation();
+                    event.preventDefault();
+                    await execute();
+                    this.update();
+                }));
+                header.appendChild(toolbarItem);
+            }
+        }
+        return {
+            header,
+            disposable
+        };
+    }
+
+    protected onAfterAttach(msg: Message): void {
+        if (!this.wrapped.isAttached) {
+            Widget.attach(this.wrapped, this.body);
+        }
+        super.onAfterAttach(msg);
+    }
+
+    protected onBeforeDetach(msg: Message): void {
+        super.onBeforeDetach(msg);
+        if (this.wrapped.isAttached) {
+            Widget.detach(this.wrapped);
+        }
+    }
+
+    protected onUpdateRequest(msg: Message): void {
+        if (this.wrapped.isAttached) {
+            this.wrapped.update();
+        }
+        super.onUpdateRequest(msg);
+    }
+
+}
+
+export namespace ViewContainerPart {
+
+    /**
+     * Make sure to adjust the `line-height` of the `.theia-view-container .part .header` CSS class when modifying this, and vice versa.
```

Comment:
Please align the doc with the CSS: https://github.com/theia-ide/theia/pull/5536/commits/8dea8fdea59e634eea9dc60e50dd433021c00d9e#diff-a83c42ebfd482eb91ce1279d274c64b1R4 or delete it. Thanks!